src/olmo_core/nn/conversion/state_mapping.py

```diff
@@ -0,0 +1,250 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Set, Tuple
+
+from olmo_core.config import StrEnum
+
+
+class TemplatePlaceholder(StrEnum):
+    LAYER = "[layer]"
+    EXPERT = "[expert]"
+
+
+@dataclass
+class StateMappingTemplate:
+    """
+    The template for a mapping state from one format to another format (e.g. OLMo Core to HF).
+    These mappings are 'templates' since they support keys and other metadata having placeholders
+    for information like the layer number or number of MoE experts. This class can be converted
+    to a `StateMapping` by providing the placeholder information.
+
+    The most standard mapping is a one-to-one state mapping, which corresponds to a single
+    string entry for both `source_template_keys` and `dest_template_keys`. The class also supports
```

Comment:
I still see a bunch of instances like this that should have double backticks. You can check how this renders by running `make docs`, though you may need to add a page for this submodule in `docs/source/nn/` (and link to it in `docs/source/nn/index.rst`).

```suggestion
    string entry for both ``source_template_keys`` and ``dest_template_keys``. The class also supports
```